packages/grafana-ui/src/types/datasource.ts

```diff
@@ -377,6 +377,13 @@ export interface DataStreamState {
    */
   delta?: DataFrame[];
 
+  /**
+   * Is the data the entire buffer, or just the changes
+   *
+   * NOTE: optional now, but should be required
+   */
+  shape?: StreamingResponseShape;
```

Comment:
Can't we infer it from the data? Like see the range and compare with existing range or something like that? If not I would rather just be for `isDelta` as I assume there cannot be more than 2 value, or can?

Reply:
The `StreamingResponseShape` right now can be `full` or `delta`. Can there be third type?

Reply:
reqesting a delta of the data => `request.isDelta` kinda works for me but just a suggestion, do not know if that can be ambiguous in some context.